application.py

```diff
@@ -77,29 +77,47 @@ def allowed_file(filename):
 
 @application.route('/api/image', methods=['POST'])
 def generate_image():
-    # check if the post request has the file part
-    if 'file' not in request.files:
-        abort(400, "No 'file' in the request.")
-    file = request.files['file']
-    # if user does not select file, browser also
-    # submit an empty part without filename
-    if file.filename == '':
-        abort(400, "The filename  in the request was empty.")
-
-    if not allowed_file(file.filename):
-        abort(400, "Bad file extension, only '.wrl' filetypes are allowed")
-
-    if file and allowed_file(file.filename):
-        # Save the uploaded file to a temporary file and then pass
-        # that on to the matlab image generation code
-        with NamedTemporaryFile() as f:
-            file.save(f.name)
-            image_fname = call_matlab_image(f.name)
-            return send_file(image_fname, mimetype='image/png')
+    # the file collection is lost when requests go through the API Gateway http proxy
+    file = request.files.get('file') or request.environ.get('wsgi.input')
+    if not file:
+        abort(411, f"file or wsgi.input argument is required\n{pprint.pformat(('REQUEST', request.environ))}")
+
+    # get the file path if possible
+    # it works for local flask mode and for direct access to EB instance
+    # it doesn't work for API Gateway http proxy in front of EB instance
```

Comment:
it does work with the API Gateway rest proxy, though.

Reply:
Interesting, was this the http Gateway that I set up before? That one never worked if that was the case, just me testing things out. I should just delete that API Gateway...

Reply:
yes, same http gateway you set up before. It seems that it loses some part of the request while being passed through.

I read that the http gateway costs 1/3 as much as the rest gateway and is faster. It works fully with the changes here, although the file extension checking is circumvented. Actually parsing wrl files will confirm they are correct, so I don't think the file extension check is essential. I'm not sure if the http gateway supports private EC2s yet, though.